lib/docker/imageservice_test.go

```diff
@@ -82,6 +83,94 @@ func (r *ImageServiceSuite) TestListsRepos(c *C) {
 	c.Assert(repos, DeepEquals, []string{"a", "b", "c", "d", "e"})
 }
 
+func (r *ImageServiceSuite) TestMergingScopeActions(c *C) {
+	cases := []struct {
+		scopes   []registryauth.RepositoryScope
+		expected []registryauth.RepositoryScope
+	}{
+		{
+			// merging multiple actions into the same scope
```

Comment:
can you turn the comment into code so failing tests can be easily identified:
```go
comment: "merging multiple actions into the same scope",
...
comment := check.Commentf(tc.comment)
c.Assert(th.scopes, DeepEquals, tt.expected, comment)
```

Reply:
This particular asset library doesn't support comments on DeepEquals (unexpected number of args)

Reply:
hmm, works for me:
```go
func (r *ImageServiceSuite) TestMergingScopeActions(c *C) {
	cases := []struct {
		comment  string
		scopes   []registryauth.RepositoryScope
		expected []registryauth.RepositoryScope
	}{
		{
			comment: "merging multiple actions into the same scope",
			scopes: []registryauth.RepositoryScope{
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"push"},
				},
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"pull"},
				},
			},
			expected: []registryauth.RepositoryScope{
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"push", "pull"},
				},
			},
		},
		{
			comment: "more that one scope added to the list without merging",
			scopes: []registryauth.RepositoryScope{
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"push"},
				},
				{
					Repository: "b",
					Class:      "b",
					Actions:    []string{"pull"},
				},
			},
			expected: []registryauth.RepositoryScope{
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"push"},
				},
				{
					Repository: "b",
					Class:      "b",
					Actions:    []string{"pull"},
				},
			},
		},
		{
			comment: "adding the same scope more than once should only include it once",
			scopes: []registryauth.RepositoryScope{
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"push"},
				},
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"push"},
				},
			},
			expected: []registryauth.RepositoryScope{
				{
					Repository: "a",
					Class:      "a",
					Actions:    []string{"push"},
				},
			},
		},
	}

	for _, tt := range cases {
		comment := Commentf(tt.comment)
		th := &multiScopeTokenHandler{}
		for _, scope := range tt.scopes {
			th.AddScope(scope)
		}

		c.Assert(th.scopes, DeepEquals, tt.expected, comment)
	}
}
```

Reply:
Ok, it would seem to be a mistake on my end, the library produces an unexpected error if the comment is passed as a string, indicating the wrong number of arguments, which I incorrectly assumed to be an issue with the library not allowing a comment.